exporter/exporterhelper/internal/persistent_storage.go

```diff
@@ -369,14 +377,31 @@ func (pcs *persistentContiguousStorage) itemDispatchingFinish(ctx context.Contex
 	}
 	pcs.currentlyDispatchedItems = updatedDispatchedItems
 
-	_, err := newBatch(pcs).
+	batch = newBatch(pcs).
 		setItemIndexArray(currentlyDispatchedItemsKey, pcs.currentlyDispatchedItems).
-		delete(pcs.itemKey(index)).
-		execute(ctx)
-	if err != nil {
-		pcs.logger.Debug("Failed updating currently dispatched items",
+		delete(pcs.itemKey(index))
+	if _, err := batch.execute(ctx); err != nil {
+		// got an error, try to gracefully handle it
+		pcs.logger.Warn("Failed updating currently dispatched items, trying to delete the item first",
 			zap.String(zapQueueNameKey, pcs.queueName), zap.Error(err))
+	} else {
+		return nil
```

Comment:
Small comment at this branch, since it is an "unnatural" to exit successfully on an else (in general).

// Everything ok, exit